File: 2134-minimum-swaps-to-group-all-1s-together-ii/2134-minimum-swaps-to-group-all-1s-together-ii.cs
public class Solution {
    public int MinSwaps(int[] nums) {
        int n = nums.Length;
        int windowSize = nums.Count(num => num == 1);
        int l = 0, r = 0, counter = 0, max = 0;
        
        for(; r < 2*n; r++){
            counter += nums[r%n];
            if(r - l + 1 > windowSize){
                counter -= nums[l%n];
                l++;
            }
            
            max = Math.Max(counter, max);
        }
        
        return windowSize - max;
    }
}

File: 2134-minimum-swaps-to-group-all-1s-together-ii/2134-minimum-swaps-to-group-all-1s-together-ii.cpp
class Solution {
public:
    int minSwaps(vector<int>& nums) {
        int n = nums.size(), windowSize = 0;
        for(int num : nums) if(num) windowSize++;
        int cntOne = 0, max = 0;
        int l = 0, r = 0;
        for(int r = 0; r < 2*n; r++){
            cntOne += nums[r%n];
            if(r-l+1 > windowSize){
                cntOne-= nums[l%n];
                l+=1;
            }
            max = fmax(cntOne, max);
        }
        return windowSize - max;
    }
};